dyno-core/src/main/java/com/netflix/dyno/connectionpool/impl/ConnectionPoolImpl.java

```diff
@@ -284,9 +284,13 @@ public boolean apply(HostConnectionPool<CL> hostPool) {
 		do  {
 			Connection<CL> connection = null;
 			
-			try { 
-					connection = 
-							selectionStrategy.getConnection(op, cpConfiguration.getMaxTimeoutWhenExhausted(), TimeUnit.MILLISECONDS);
+			try {
+					connection = selectionStrategy.getConnectionUsingRetryPolicy(
+                            op,
```

Comment:
I think the tabs messed up the rendering; changed the file indent with spaces now.